pmpro-woocommerce.php

```diff
@@ -407,56 +407,65 @@ function pmprowoo_get_membership_price( $price, $product ) {
 		return $price;
 	}
 	
-	$discount_price = $price;
-	
 	$membership_product_ids = array_keys( $pmprowoo_product_levels );
 	$items       = is_object( WC()->cart ) ? WC()->cart->get_cart_contents() : array(); // items in the cart
 	
 	//ignore membership products and subscriptions if we are set that way
-	if ( ( ! $pmprowoo_discounts_on_subscriptions || $pmprowoo_discounts_on_subscriptions == 'No' ) && ( $product->get_type() == "subscription" || $product->get_type() == "variable-subscription" || in_array( $product->get_id(), array_keys( $pmprowoo_product_levels ), false ) ) ) {
```

Comment:
This breaks when you want to stop membership linked products from being discounted based on the advanced setting.

The `$product->get_id` doesn't work for variations, so we should figure out the product type before this and then use `->get_id` or `->get_parent_id`.

Reply:
The setting under "Advanced Settings" > "Apply Member Discounts to WooCommerce Subscription and Membership Products?"

Reply:
This would affect all membership products that are not subscription products. Steps to recreate:

1. Create a simple product in Woo and set the Membership Level that is given whenever purchased.
2. Toggle the Advanced setting and view the product each time.